unit_test/test_build_ids.py

```diff
@@ -0,0 +1,28 @@
+import toml
+from toml.encoder import TomlEncoder
+
+from cibuildwheel.util import resources_dir
+
+
+class InlineArrayDictEncoder(TomlEncoder):
```

Comment:
Hah, wow!

Reply:
Yeah, the libraries for TOML are not, um, amazing examples of great libraries. There was supposed to be a way to preserve inline status but it doesn't work for inline arrays, the beginning of the work seems to be there from several years back, but unfinished. So we can just handle it here. Or use JSON. Though we'd probably have to control the output a bit there too to keep it nice.

Reply:
Still nice that it's possible.

However, seeing this: is it very important that we can roundtrip?

Reply:
Oh, of course, if we want to automate this in the next step, it is :-)